src/main.rs

```diff
@@ -14,14 +16,20 @@ async fn main() {
 
     let config = config::Config::new();
     let addr = SocketAddr::from(([127, 0, 0, 1], config.server_port));
-    let context = context::Context::new(config).await;
+    let context = context::Context::new(&config).await;
 
     context.bootstrap().await;
 
     let app = Router::new()
         .route("/:hash", get(handlers::redirect::redirect))
         .route("/new", post(handlers::create_link::create_link))
-        .layer(Extension(context));
+        .layer(Extension(context))
+        .layer(
+            CorsLayer::new()
+                .allow_origin(config.cors_allow_origin.parse::<HeaderValue>().unwrap())
+                .allow_headers([header::CONTENT_TYPE])
+                .allow_methods([Method::GET, Method::POST]),
+        );
```

Comment:
For some users CORS is not suitable for their use case. To allow these users to remove CORS from their instance it would be nice to provide CORS optionally using for instance: a feature flag or environment variables configuration. This is still a discussion to have and requires some research.